### DISCUSSION TITLE
Is there a way to keep the original file name for attached media?

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#10538] [Q&A] [answered]
I have a content type with a media field that is set to use the "Attached" editor.  When I add files to a content item the file name gets replaced with a unique id which makes sense but I need to keep the original file name somewhere to display to the users.  Is there a way to have the original file name stored along with the unique name?  

I thought about using the MediaText property for that and making my own editor by copying  `MediaField-Attached.Edit.cshtml` from the Media module and changing it to automatically set the MediaText to the original file name but I don't see how I could set it there.

**Comment by DrewBrasher** [comment, marked as answer]
Issue https://github.com/OrchardCMS/OrchardCore/issues/4563 is the same thing I'm asking.

**Reply by hishamco** [reply]
I was looking for that issue to link it here, but you catch it first ;)
